src/main/java/org/jenkins/plugins/lockableresources/LockableResource.java

```diff
@@ -168,7 +184,25 @@ public String getReservedBy() {
 	}
 
 	@Exported
-	public boolean isReserved() {
+	public boolean isReserved() {	
+		
+		StratOSControllerAPI stratosAPI = LockableResourcesManager.get().getControllerAPI();
+		List<Objective> reservationObjectives = stratosAPI.objectives().getByType("SUDS/1.0/reservation");
+		String selfUid = getUid();
+		
+		if(reservationObjectives.isEmpty()){
```

Comment:
This seems redundant. Don't you just want to execute the code in the else block if !reservationObjectives.isEmpty. reservedBy will be null so you can just fall through to the return with the same result.